### DISCUSSION TITLE
Add single-file mode for pull requests

### DISCUSSION THREAD

**Discussion by nikeshparajuli** [EnixCoda/Gitako#205] [Ideas]
First of all, thank you for the great extension! I had a some questions about the features...wasn't sure if they exist already or are in the process of being implemented.

Here are my questions:

1. Is there a way to view only one file at a time when looking at Files Changed in a PR? So when a user goes to the Files Changed tab, by default, the first modified file in the tree should be selected and shown on the side. The other files shouldn't be shown (the way GitHub shows all the files changed in a single page really gives me a panic attack every time 😄). This would make things much cleaner. I do appreciate the highlight around a selected file though (this shouldn't be required if only one file is shown).

Also, just noticed that for files like .gitignore and others that live in the root of the repo, they are shown first by GitHub on Files Changed. When looking at the tree this feels a little confusing as it shows the folders and their changed files first and the root files at the end (as it should be). Showing only one file at a time should remove this ambiguity.

2. When viewing a commit, can the extension only show the files that were changed (like in the PR, along with the features suggested above) instead of the main repo tree?

I saw some extensions that could do one of the above things but not both. I feel that this would be a perfect addition to Gitako if they aren't present already. I thank you your time!

**Comment by EnixCoda** [comment]
Hi @nikeshparajuli Thank you for the questions.

1. Sorry that I do not see the point of rendering only one file.
If only one file is rendered, then what functions do you expect Gitako to provide, like navigating to the next file? But Gitako does not know which the next file is, it has to list all the changed files for users to choose. If the complete list of changed files annoys you when reviewing a single file, how about collapsing the sidebar (by default it should collapse when you move your cursor out of it)?
Let me know if there is anything I'm missing from your questions. :)

2. This is a good idea. But I do not need this feature much. I can implement it if more users request it.

**Reply by nikeshparajuli** [reply]
For # 1, I meant something like the first screenshot below. I didn't mean that the extension/Gitako tree annoyed me. The Gitako tree should still show all the files changed but should only render one file at a time on the right panel based on what the user selects. It's the GitHub's way of rendering all files at once that gives me the discomfort. 😄 

![image](https://user-images.githubusercontent.com/47485043/153629217-3abdcf64-1e70-450d-8f07-c8378f95d4e8.png)

vs

![image](https://user-images.githubusercontent.com/47485043/153629254-6f0be8fd-77d3-4f82-b4e1-6c7ab7ed6481.png)

The first screenshot is from the [GitHub Diff Explorer](https://chrome.google.com/webstore/detail/github-diff-explorer/kagcmhcnjehpeihgmcohmdceffihkglk/related?hl=en-US) extension. Let me know if this makes sense.

**Reply by EnixCoda** [reply]
Oh, I see! Thank you! It looks quite useful.

**Reply by nikeshparajuli** [reply]
It sure does! Hope to see it implemented soon. :)

**Comment by EnixCoda** [comment]
Updating the title as the view-commit-files feature has been released
